nexus/src/app/vpc_router.rs

```diff
@@ -129,9 +129,9 @@ impl super::Nexus {
         // router kind cannot be changed, but it might be able to save us a
         // database round-trip.
         if db_router.kind == VpcRouterKind::System {
-            return Err(Error::MethodNotAllowed {
-                internal_message: "Cannot delete system router".to_string(),
-            });
+            return Err(Error::method_not_allowed(
```

Comment:
I'm not sure if this would break anything but I think I'd make this a 403 or 400 instead and remove the MethodNotAllowed variant.  405 feels like a lower-level error -- I'd expect a client to be pretty confused by that.  This is just my intuition -- I checked docs and I don't think 405 is necessarily wrong here.

Reply:
Yeah, agreed. Not providing the "Allow" header was the clincher for me. Turned this into 400.